### PR TITLE
Add subheadline proptype, reducing log noise

### LINKED ISSUE #848: Subheadlines fail the mainContent proptypes
Describe the bug
Subheadlines currently don't have a proptype, & are not included in the main content proptypes, resulting in unnecessary & distracting logging on dev when loading a page with a subheadline.

To Reproduce

**Reproduction Steps**
Run `npm run dev` or enter `rs` in an existing session -- the issue will only appear on first run.
Visit http://localhost:7080/news/articles/c9rpqy7pmypo (To be updated)
View the terminal to see the error.

The issue does not appear in `c85pqyj5m2ko`, as it contains no subheadline blocks.

screen shot 2018-10-25 at 16 51 16

Expected behavior
Subheadline should be considered a valid block, & not cause unnecessary console warnings.

### PR DESCRIPTION
**Pull request by bcmn**
Resolves #848.

Opened to reduce logging noise. Currently, requesting scenario 25 results in a console log, as it contains a block with the type `subheadline`, for which no block proptype exists.

screen shot 2018-10-25 at 16 51 16

As `subheadline` has exactly the same proptype requirements as `headline`, this PR allows block proptype creations to specify an array of applicable `type` strings. 

[For example:](https://github.com/BBC-News/simorgh/blob/add_subheadline_proptype/src/app/models/propTypes/headline/index.js#L8-L12)
```
export const headlineBlockPropTypes = blockOfValidTypeAndModel(
  ['headline', 'subheadline'],
  headlineModelPropTypes,
);
``` 

---

**Reproduction Steps**
Run `npm run dev` or enter `rs` in an existing session -- the issue will only appear on first run.
Visit http://localhost:7080/news/articles/c9rpqy7pmypo (To be updated)
View the terminal to see the error.

The issue does not appear in `c85pqyj5m2ko`, as it contains no subheadline blocks.

- [ ] Tests added for new features
- [ ] Test engineer approval

### CODE REVIEW

File: src/app/models/propTypes/general/index.js [hunk 1]
```diff
@@ -1,13 +1,13 @@
 import { shape, oneOf, arrayOf, oneOfType, checkPropTypes } from 'prop-types';
 import getMissingRequiredProps from './getMissingRequiredProps';
 
-export const blockObjectOfSpecificTypeAndModel = (type, model) => ({
-  type: oneOf([type]).isRequired,
+export const blockObjectOfValidTypeAndModel = (validTypes, model) => ({
```
Comment (bcmn) [inline src/app/models/propTypes/general/index.js]: Just to clarify: This identifier was changed in an attempt to communicate that the block object is now expected to match _one of an array_ of type strings, as opposed to match _one, specific_ type string. 

I wasn't hugely satisfied by either phrasing to capture this, so I'd be keen on suggestions.

File: src/app/models/propTypes/general/index.js [hunk 2]
```diff
@@ -1,13 +1,13 @@
...
+  type: oneOf(validTypes).isRequired,
   model: shape(model).isRequired,
 });
 
-export const blockOfSpecificTypeAndModel = (type, model) =>
-  shape(blockObjectOfSpecificTypeAndModel(type, model));
+export const blockOfValidTypeAndModel = (validTypes, model) =>
```
Comment (bcmn) [inline src/app/models/propTypes/general/index.js]: [See above comment on changing identifiers.](https://github.com/BBC-News/simorgh/pull/845/files#r228883507)

File: src/app/models/propTypes/general/index.js (hunk 1 above)
Comment (jtart) [reply src/app/models/propTypes/general/index.js]: I think it's good to make it more identifiable, what about just `types`? The function could also be in the plural, i.e. `blockObjectOfTypesAndModel`

Comment (bcmn) [reply src/app/models/propTypes/general/index.js]: Renamed to take this into account. Cheers 😊

**Review by jtart — APPROVED** [review]

**Review by radiocontrolled — APPROVED** [review]

**Pull request merged**